src/services/auth.service.ts

```diff
@@ -0,0 +1,34 @@
+import {/* inject, */ BindingScope, injectable} from '@loopback/core';
+import {repository} from '@loopback/repository';
+import {performance} from 'perf_hooks';
+import {AuthRepository} from '../repositories';
+
+@injectable({scope: BindingScope.TRANSIENT})
+export class AuthService {
+  constructor(
+    @repository(AuthRepository) public apiRepository: AuthRepository,
+  ) {}
+
+  /*
+   * Adapted from https://stackoverflow.com/a/8809472/8396479
+   */
+  generateApiKey() {
```

Comment:
why not using uuid module here ? (it's even mentioned in your SO post referenced)
the other use could be crypto.randomUUID()

But let's not reinvent the wheel here